File: tests/fixtures/multi_lang_repo/module.ts
// TypeScript file in multi-language repository

/**
 * TypeScript interface
 */
export interface Data {
  id: number;
  name: string;
}

/**
 * TypeScript function with typing
 */
export function tsFunction(data: Data): string {
  return `ID: ${data.id}, Name: ${data.name}`;
}

/**
 * TypeScript class
 */
export class TSClass {
  private value: number;

  constructor(value: number) {
    this.value = value;
  }

  getValue(): number {
    return this.value;
  }
}

File: tests/fixtures/multi_lang_repo/module.cs
// C# file in multi-language repository

namespace MultiLang
{
    public class CSharpClass
    {
        public int Value { get; set; }

        public CSharpClass(int value)
        {
            Value = value;
        }

        public int GetValue()
        {
            return Value;
        }
    }

    public static class CSharpHelper
    {
        public static int CSharpFunction(int x)
        {
            return x * 2;
        }
    }
}
